### PR TITLE
Fetch PractitionerId, PractitionerLocation, PractitionerCareTeam, PractitionerOrganization from Shared Preferences

### LINKED ISSUE #2900: Add functions to fetch Assignment data using Rules on the Configs
**Describe the feature request.**
- The OpenSRP app stores assignment data in the secure shared preferences. We would like to write some methods on the  [RulesFactory](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt) to allow the fetching of this assignment data via rules on the configs
- Each assignment data point should be fetched separately

**Acceptance criteria**
- [ ] Define methods that allow the fetching of the following assignment data
    - [ ] Practitioner Id
    - [ ] Practitioner Assigned CareTeam(s)
    - [ ] Practitioner Assigned Organization(s)
    - [ ] Practitioner Assigned Location(s) 

**Implementation plan (For Engineers)**
The plan for implementing the solution e.g. via a _description_ or a _check list_ for the various ordered tasks that will need to be completed.
_i.e. Describe how you intend to solve the problem_

### PR DESCRIPTION
**Pull request by Raynafs**
Extract PractitionerId, PractitionerLocation, PractitionerCareTeam, PractitionerOrganization
    * Create new enum class called PractitionerKey
    * Create a function called extractSharedPrefValues in RulesFactory

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2900 
Depends on PR #2873 (Which saves practitioner details on shared preferences)

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [x] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [x] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by SebaMutuku** [comment]
@Raynafs it would be nice if you added documentation on how to use this particular rule

**Comment by Rkareko** [comment]
> @Raynafs it would be nice if you added documentation on how to use this particular rule

Documentation is part of the implementation plan for the [issue](https://github.com/opensrp/fhircore/issues/2900)

File: docs/engineering/android-app/configuring/working-with-rules.mdx [hunk 1]
```diff
@@ -89,3 +89,21 @@ This rule is used within the ruels engine or workflow to generate the correct se
       ]
     }
 ```
+
+## Using the `extractPractitionerInfoFromSharedPrefs()` rule
+
+This rule is used within the rules engine or workflow to extract `PractitionerId`, `PractitionerCareTeam`, `PractitionerOrganization` and `PractitionerLocation` from `Shared Preference`.
```
Comment (Rkareko) [inline docs/engineering/android-app/configuring/working-with-rules.mdx:95]: List all the enums that are supported by this change e.g in order to fetch `PractitionerId` which enum would be used?

A tabular format would be easy to read.

Comment (Raynafs) [reply docs/engineering/android-app/configuring/working-with-rules.mdx:95]: Sure. I have done so

**Pull request merged**